libsql/src/replication/mod.rs

```diff
@@ -256,13 +256,19 @@ impl EmbeddedReplicator {
             }
         }
 
-        let last_frames_synced = self.last_frames_synced.fetch_add(
-            replicator.frames_synced(),
-            std::sync::atomic::Ordering::Relaxed,
-        );
+        let current_frames_synced = replicator.frames_synced();
+
+        let last_frames_synced = self
+            .last_frames_synced
+            .load(std::sync::atomic::Ordering::Relaxed);
+
+        if current_frames_synced > last_frames_synced {
+            self.last_frames_synced
```

Comment:
This is incorrect for concurrent updates. You need to use compare and set

Reply:
Switched it to `compare_and_exchange`, is that what you're looking for?